tools/stress-cluster/scripts/deploy_stress_tests.ps1

```diff
@@ -0,0 +1,160 @@
+[CmdletBinding(DefaultParameterSetName = 'Default')]
+param(
+    [string]$SearchDirectory,
+    [hashtable]$Filters,
+    [string]$Environment,
+    [string]$Repository,
+    [switch]$PushImages,
+    [string]$ClusterGroup,
+    [string]$DeployId,
+
+    [Parameter(ParameterSetName = 'DoLogin', Mandatory = $true)]
+    [switch]$Login,
+
+    [Parameter(ParameterSetName = 'DoLogin')]
+    [string]$Subscription
+)
+
+$ErrorActionPreference = 'Stop'
+
+$FailedCommands = New-Object Collections.Generic.List[hashtable]
+
+. $PSScriptRoot/find_all_stress_packages.ps1
+
+# Powershell does not (at time of writing) treat exit codes from external binaries
+# as cause for stopping execution, so do this via a wrapper function.
+# See https://github.com/PowerShell/PowerShell-RFC/pull/277
+function Run() {
+    Write-Output "" "==> $args" ""
+    $command, $arguments = $args
+    & $command $arguments
+    if ($LASTEXITCODE) {
+        Write-Error "Command '$args' failed with code: $LASTEXITCODE" -ErrorAction 'Continue'
+        $FailedCommands.Add(@{ command = "$args"; code = $LASTEXITCODE })
+    }
+}
+
+function RunOrExit() {
+    run @args
+    if ($LASTEXITCODE) {
+        exit $LASTEXITCODE
+    }
+}
+
+function Login([string]$subscription, [string]$clusterGroup, [boolean]$pushImages) {
+    Write-Output "Logging in to subscription, cluster and container registry"
+    az account show
```

Comment:
If you want to try and hide the extra output here you might want to consider a pattern like https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Prepare-Release.ps1#L150